domain-services/fiscal-event-service/src/main/java/org/egov/service/FiscalEventService.java

```diff
@@ -53,81 +50,27 @@ public class FiscalEventService {
         enricher.enrichFiscalEventPushPost(fiscalEventRequest);
 
-        FiscalEventRequest fiscalEventRequestCopy = new FiscalEventRequest();
-        BeanUtils.copyProperties(fiscalEventRequest, fiscalEventRequestCopy);
-        //nullify the coaCode before pushing to topics
-        nullifyCoaCodeAndAddVersion(fiscalEventRequest);
-
-        if (fiscalEventRequest.getFiscalEvent() != null && !fiscalEventRequest.getFiscalEvent().isEmpty()) {
-            RequestHeader requestHeader = fiscalEventRequest.getRequestHeader();
-            for (FiscalEvent fiscalEvent : fiscalEventRequest.getFiscalEvent()) {
-                ObjectNode fiscalEventRequestNode = JsonNodeFactory.instance.objectNode();
-                fiscalEventRequestNode.putPOJO("requestHeader", requestHeader);
-                fiscalEventRequestNode.putPOJO("fiscalEvent", fiscalEvent);
-
-                //push with request header details
-                producer.push(eventConfiguration.getFiscalPushRequest(), fiscalEventRequestNode);
-                //push without request header details
-                producer.push(eventConfiguration.getFiscalEventPushToMongoSink(), fiscalEvent);
-            }
-        }
-        //nullify coaId before sending the response
-        nullifyCoaIdCodeAndAddVersion(fiscalEventRequestCopy);
-        return fiscalEventRequestCopy;
-    }
+        if (!CollectionUtils.isEmpty(fiscalEventRequest.getFiscalEvent())) {
+            //push with request header details
+            producer.push(eventConfiguration.getFiscalPushRequest(), fiscalEventRequest);
 
-    private void nullifyCoaIdCodeAndAddVersion(FiscalEventRequest fiscalEventRequest) {
-        List<FiscalEvent> fiscalEvents = new ArrayList<>();
-
-        if (fiscalEventRequest.getFiscalEvent() != null && !fiscalEventRequest.getFiscalEvent().isEmpty()) {
-            for (FiscalEvent fiscalEvent : fiscalEventRequest.getFiscalEvent()) {
-                FiscalEvent fiscalEventCopy = new FiscalEvent();
-                BeanUtils.copyProperties(fiscalEvent, fiscalEventCopy);
-
-                if (fiscalEvent.getAmountDetails() != null && !fiscalEvent.getAmountDetails().isEmpty()) {
-                    List<Amount> amounts = new ArrayList<>();
-                    for (Amount amount : fiscalEvent.getAmountDetails()) {
-                        Amount amountCopy = new Amount();
-                        BeanUtils.copyProperties(amount, amountCopy);
-                        amountCopy.setCoaId(null);
-                        amounts.add(amountCopy);
-                    }
-                    fiscalEventCopy.setVersion(FISCAL_EVENT_VERSION);
-                    fiscalEventCopy.setAmountDetails(amounts);
-                }
-                fiscalEvents.add(fiscalEventCopy);
-            }
-            fiscalEventRequest.setFiscalEvent(fiscalEvents);
-        }
-    }
+            //push to ES sink
+            ProducerPOJO objectForEsSink = new ProducerPOJO();
```

Comment:
Make this work without the new ProducerPOJO class. 
Just putting fiscalEventRequest should work

Reply:
Implemented in this service.
Pending updates in ifix-es-pipeline and ifix-migration-toolkit.